### PR TITLE
Add Jedis pool metrics binder

### PR DESCRIPTION
**Pull request by pditommaso**
This PR adds a Micrometer metrics binder to export Jedis pool via Promeheus. 

Tagging @swampie for visibility

### CODE REVIEW

**Comment by pditommaso** [comment]
@munishchouhan this adds an Prometheus exporter for the Redis connection metrics. Ca =n you please validate and add the charts in the Redis section in Grafana?

File: src/main/groovy/io/seqera/wave/redis/JedisPoolMetricsBinder.groovy [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        // Borrow/Return statistics
+        registry.gauge("jedis.pool.borrowed", pool, JedisPool::getBorrowedCount);
+        registry.gauge("jedis.pool.returned", pool, JedisPool::getReturnedCount);
+
+        // Additional metrics (resets, evictions, etc.)
+        registry.gauge("jedis.pool.max.borrow.wait.millis", pool, (p)-> p.maxBorrowWaitDuration.toMillis() as double)
+        registry.gauge("jedis.pool.mean.borrow.wait.millis", pool, (p)-> p.meanBorrowWaitDuration.toMillis() as double)
+        registry.gauge("jedis.pool.mean.active.millis", pool, (p)-> p.meanActiveDuration.toMillis() as double)
+        registry.gauge("jedis.pool.mean.idle.millis", pool, (p)-> p.meanIdleDuration.toMillis() as double)
+
```
Comment (pditommaso) [inline src/main/groovy/io/seqera/wave/redis/JedisPoolMetricsBinder.groovy:59]:
```suggestion
```

**Comment by munishchouhan** [comment]
Validated:
I can see all the Jedispool metrics being scaped by Prometheus in my local machine:
Screenshot 2024-12-04 at 07 22 07

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-1.15.4
+1.15.5-B2
```
Comment (pditommaso) [inline VERSION:1]:
```suggestion
1.15.4
```

**Comment by pditommaso** [comment]
Nice. Is there already a PR for the Grafana dashboard

**Comment by munishchouhan** [comment]
> Nice. Is there already a PR for the Grafana dashboard?

Working on it

**Pull request merged**